apps/cli/commands/list_documents.py

```
@@ -0,0 +1,141 @@
+        if result.total > result.limit:
+            pages = (result.total + result.limit - 1) // result.limit
+            current_page = (result.offset // result.limit) + 1
+            console.print(
+                f"\n[dim]Page {current_page} of {pages} "
+                f"(showing {result.offset + 1}-{result.offset + len(result.documents)} "
+                f"of {result.total})[/dim]"
+            )
+
+    except ValueError as e:
+        console.print(f"[red]Error:[/red] {e}")
+        raise typer.Exit(1)
```

Ruff (0.14.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)